app/models/event.rb

```diff
@@ -17,16 +18,25 @@ def coordinates
   end
 
   def self.by_name(params)
-    where("name LIKE ?", "%#{params[:query]}%").map{ |event| EventPresenter.new(event)}
+    results = where("name LIKE ?", "%#{params[:query]}%")
+    to_presenter(results)
   end
 
   def self.by_date(params)
-    date = params[:query].to_date
-    where(:date => date.beginning_of_day..date.end_of_day).map{ |event| EventPresenter.new(event)}
+    start_date = params[:query].to_date
+    end_date = params[:end_date].to_date
+    results = where(:date => start_date.beginning_of_day..end_date.end_of_day)
+    to_presenter(results)
   end
 
   def self.by_location(params)
-    near(params[:location], params[:radius]).map{ |event| EventPresenter.new(event)}
```

Comment:
I like how we are using the EventPresenter across the entire app for different features. It worked out great!